mobile/lib/shared/providers/asset.provider.dart

```diff
@@ -52,53 +66,67 @@ class AssetNotifier extends StateNotifier<List<AssetResponseDto>> {
   }
 
   onNewAssetUploaded(AssetResponseDto newAsset) {
-    state = [...state, newAsset];
+    state = [...state, Asset.remote(newAsset)];
```

Comment:
This is called by the ws connection. Maybe the duplication occurs because we don't compare the id of the new asset agains the ids of our local assets here?

Reply:
yeah, that makes sense!

This is sort of non-trivial to fix without ruining upload/backup performance on large collections. I need to think about either making the state a HashSet or aggregating  uploaded assets and updating only in batches after a certain timeout etc.

Reply:
fixed: hope I found a good solution